pkg/conversion/mapi2capi/openstack.go

```diff
@@ -0,0 +1,789 @@
+
+			capoProfile.OVSHWOffload = &capoOVSHWOffload
+		}
+
+		if _, ok := mapoPort.Profile["trusted"]; ok {
+			capoTrustedVF := false
+			// TODO(stephenfin): Does neutron allow other "truthy" values?
+			if mapoPort.Profile["trusted"] == "true" {
+				capoTrustedVF = true
+			}
+
+			capoProfile.TrustedVF = &capoTrustedVF
+		}
+		// LOSSY!! We don't/can't handle other profile flags.
+		capoPort.Profile = &capoProfile
+
+		// convert .SecurityGroups
+		capoSecurityGroups := []capov1.SecurityGroupParam{}
+
+		if mapoPort.SecurityGroups != nil {
+			for _, mapoSecurityGroup := range *mapoPort.SecurityGroups {
+				capoSecurityGroup := capov1.SecurityGroupParam{
+					ID: &mapoSecurityGroup,
+				}
+				capoSecurityGroups = append(capoSecurityGroups, capoSecurityGroup)
+			}
+		}
+
+		if len(capoSecurityGroups) > 0 {
+			capoPort.SecurityGroups = capoSecurityGroups
+		}
+
+		// We intentionally ignore the DeprecatedHostID field since it's now ignored by
+		// MAPO itself.
+		if mapoPort.DeprecatedHostID != "" {
+			warnings = append(warnings, field.Invalid(fldPath.Index(i).Child("hostID"), mapoPort.DeprecatedHostID, "hostID is ignored by MAPO, ignoring").Error())
+		}
+
+		capoPorts = append(capoPorts, capoPort)
+	}
+
+	return capoPorts, warnings, errors
+}
+
+func convertMAPORootVolumeToCAPO(fldPath *field.Path, mapoRootVolume *mapiv1alpha1.RootVolume) (*capov1.RootVolume, []string, field.ErrorList) {
+	errors := field.ErrorList{}
+	warnings := []string{}
+
+	if mapoRootVolume == nil {
+		return nil, warnings, errors
+	}
+
+	capoRootVolume := &capov1.RootVolume{}
+	// TODO(stephenfin): CAPO uses GiB, MAPO allegedly uses GB. Are they actually different (and therefore need conversion)?
+	capoRootVolume.SizeGiB = mapoRootVolume.Size
+	capoRootVolume.Type = mapoRootVolume.VolumeType
+
+	if mapoRootVolume.Zone != "" {
+		name := capov1.VolumeAZName(mapoRootVolume.Zone)
+		capoRootVolume.AvailabilityZone = &capov1.VolumeAvailabilityZone{
+			From: "Name",
+			Name: &name,
+		}
+	}
+
+	// We intentionally ignore the DeprecatedSourceType, DeprecatedDeviceType fields since they're
+	// now ignored by MAPO itself and they have no equivalent in CAPO
+	if mapoRootVolume.DeprecatedDeviceType != "" {
+		// deviceType is deprecated and silently ignored.
+		// See
+		warnings = append(warnings, field.Invalid(fldPath.Child("deviceType"), mapoRootVolume.DeprecatedDeviceType, "deviceType is silently ignored by MAPO and will not be converted").Error())
+	}
+
+	if mapoRootVolume.DeprecatedSourceType != "" {
+		// sourceType is deprecated and silently ignored.
+		// See
+		warnings = append(warnings, field.Invalid(fldPath.Child("sourceType"), mapoRootVolume.DeprecatedSourceType, "sourceType is silently ignored by MAPO and will not be converted").Error())
+	}
+
+	if mapoRootVolume.SourceUUID != "" {
+		// SourceUUID is ignored if the property is set on the platform instead.
+		// See
+		// TODO(stephenfin): We should return this value and use it if spec.image is not set
```

Comment:
this is a very old and legacy parameter from my knowledge. I know we can't remove it but I wouldn't bother too much about it.

Reply:
I changed this to a `NOTE`. We might need to rework if testing shows it's necessary.